### PR TITLE
issue-77: added logic to turn headers into atoms, covered it with tes…

### PR DESCRIPTION
**Pull request by vad2der**
…ts, expanded documentation

**This PR addresses**
Headers atomization
[issue 77](https://github.com/beatrichartz/csv/issues/77)

**Open questions**
- Anything that you wanted to discuss

**Checklist**
- Tests added
Yes, for running only relevant tests: `mix test --only atomize_headers`
- Coverage increases or stays the same
Increases
- Docs updated
Yes, in **_CSV.Decoding.Decoder_** module and in **_README.md_** files.
- Changelog updated
No, since changelog to be updated when release happens

### CODE REVIEW

File: lib/csv/decoding/decoder.ex [hunk 1]
```diff
@@ -174,7 +193,30 @@ defmodule CSV.Decoding.Decoder do
...
     case parse_row({line, 0}, options) do
       {:ok, headers, _} ->
-        {[], {headers, options}}
+        case options[:atomize_headers] do
+          :safe ->
+            {
+              [],
+              {
+                headers
+                |> Enum.map(fn (header) -> String.to_existing_atom(header) end),
```
Comment (hauleth) [inline lib/csv/decoding/decoder.ex:202]:
```suggestion
                Enum.map(headers, &String.to_existing_atom/1),
```

File: lib/csv/decoding/decoder.ex [hunk 2]
```diff
@@ -174,7 +193,30 @@ defmodule CSV.Decoding.Decoder do
...
+                options
+              }
+            }
+          
+          :unsafe ->
+            {
+              [],
+              {
+                headers
+                |> Enum.map(fn (header) -> String.to_atom(header) end),
```
Comment (hauleth) [inline lib/csv/decoding/decoder.ex:212]:
```suggestion
                Enum.map(headers, fn &String.to_atom/1),
```

File: lib/csv/decoding/decoder.ex [hunk 3]
```diff
@@ -174,7 +193,30 @@ defmodule CSV.Decoding.Decoder do
...
+                options
+              }
+            }
+          
+          _ ->
+            {[], {headers, options}}
+        end
```
Comment (hauleth) [inline lib/csv/decoding/decoder.ex:219]:
```suggestion
        transform_headers =
          case options[:atomize_headers] do
            :safe -> &Enum.map(&1, &String.to_existing_atom/1)
            :unsafe -> &Enum.map(&1, &String.to_atom/1)
            _ -> &(&1)
          end
        {[], {transform_headers.(headers), options}}
```

**Review by hauleth — COMMENTED** [review]
In general I avoid using pipeline operator when having only one step in pipeline. I see such approach overused there.

**Comment by beatrichartz** [comment]
See comment in #77 , closing this.

**Pull request closed without merging**